### DISCUSSION TITLE
Cannot login in webview (tauri)

### DISCUSSION THREAD

**Discussion by vincenzoml** [wavezync/vue3-google-signin#14] [Q&A] [answered]
I'm trying vue3-google-signin in vue3 using tauri as a backend, but this does not work. The browser windows for authentication is opened in the embedded browser and that is disallowed by google policies, so I get an error message as in the attached screenshot. 

<img width="374" alt="image" src="https://user-images.githubusercontent.com/2283125/190843555-30345ca1-9074-4e54-b605-fb240ba140b6.png">

**Comment by kasvith** [comment, marked as answer]
This is not a problem with this library.

It seems a problem with tauri's browser API. AFAIK it uses a native web view to build the app. This native web view might be outdated from google's perspective.

The best way to handle this is to use **redirect** flow by setting `ux_mode` to `redirect`  and opening up the system browser to handle the authentication flow. After that, you might want to register some handler with the system to open up Tauri app and pass the obtained token. In electron, this is done using a [protocol](https://www.electronjs.org/docs/latest/api/protocol).
Im not sure what to be used with Tauri.

This is how VSCode authenticates with GitHub also

**Reply by kasvith** [reply]
Hi @vincenzoml check this out https://vue3-google-signin.syetalabs.io/guide/using-with-electron.html

**Comment by vincenzoml** [comment]
Thanks a lot for the hint. Just a question: is `ux_mode` a parameter of a function? Which one?

**Reply by kasvith** [reply]
its availiable on both `useCodeClient` and `useTokenClient` composables.
Also its available on [`GoogleSignInButton`](https://vue3-google-signin.syetalabs.io/components/google-signin-button.html#ux-mode) 

I think I will update the documentation about this part. If you use VSCode it should probably hint also

**Comment by vincenzoml** [comment]
If I use `ux-mode="redirect"` I get no popup, but the authentication form is still opened in the same webview window as the application. No external browser is opened. Maybe there's a way to retrieve the correct login url instead (and then use a system call to open the browser?).

**Reply by kasvith** [reply]
Aha, it seems to google doesn't expose it by default. If you can wait couple of days maybe I will add another feature to get the redirect URL without actually redirecting browser 

this would be useful to anyone who wants to use a platform like Electron or Tauri

**Comment by vincenzoml** [comment]
Thanks a lot, awaiting :)

**Reply by vincenzoml** [reply]
Then one needs to open a port or receive anyway the login response at some public address? Isn't it difficult without a permanent IP?

**Reply by kasvith** [reply]
You can use tauri's server on localhost to handle this code response from google and get tokens.

**Comment by kasvith** [comment]
Tracking here https://github.com/syetalabs/vue3-google-signin/issues/15
